File: app/src/main/java/de/westnordost/streetcomplete/quests/internet_access/AddInternetAccessForm.java
package de.westnordost.streetcomplete.quests.internet_access;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RadioGroup;

import de.westnordost.streetcomplete.R;
import de.westnordost.streetcomplete.quests.AbstractQuestFormAnswerFragment;

public class AddInternetAccessForm extends AbstractQuestFormAnswerFragment
{
	private RadioGroup radioGroup;

	public static final String INTERNET_ACCESS = "internet_access";

	private static final String
			WLAN = "wlan",
			NO = "no",
			TERMINAL = "terminal",
			WIRED = "wired";

	@Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
							 Bundle savedInstanceState)
	{
		View view = super.onCreateView(inflater, container, savedInstanceState);
		View contentView = setContentView(R.layout.quest_internet_access);
		radioGroup = contentView.findViewById(R.id.radioButtonGroup);
		radioGroup.setOnCheckedChangeListener((group, checkedId) -> checkIsFormComplete());
		return view;
	}

	@Override protected void onClickOk()
	{
		Bundle answer = new Bundle();
		answer.putString(INTERNET_ACCESS, getInternetAccessValueByCheckedRadioButtonId());
		applyAnswer(answer);
	}

	private String getInternetAccessValueByCheckedRadioButtonId()
	{
		switch (radioGroup.getCheckedRadioButtonId())
		{
			case R.id.wlan:      return WLAN;
			case R.id.no:        return NO;
			case R.id.terminal:  return TERMINAL;
			case R.id.wired:     return WIRED;
		}
		return null;
	}

	@Override public boolean isFormComplete() { return radioGroup.getCheckedRadioButtonId() != -1; }
}

File: app/src/main/java/de/westnordost/streetcomplete/quests/internet_access/AddInternetAccessForm.kt
package de.westnordost.streetcomplete.quests.internet_access

import de.westnordost.streetcomplete.R
import de.westnordost.streetcomplete.quests.AListQuestForm
import de.westnordost.streetcomplete.quests.TextItem
import de.westnordost.streetcomplete.quests.internet_access.InternetAccess.NO
import de.westnordost.streetcomplete.quests.internet_access.InternetAccess.TERMINAL
import de.westnordost.streetcomplete.quests.internet_access.InternetAccess.WIFI
import de.westnordost.streetcomplete.quests.internet_access.InternetAccess.WIRED

class AddInternetAccessForm : AListQuestForm<InternetAccess>() {

    override val items = listOf(
        TextItem(WIFI, R.string.quest_internet_access_wlan),
        TextItem(NO, R.string.quest_internet_access_no),
        TextItem(TERMINAL, R.string.quest_internet_access_terminal),
        TextItem(WIRED, R.string.quest_internet_access_wired),
    )
}
